src/components/forms/components/DragAndDropUpload.js

```diff
@@ -0,0 +1,674 @@
+                }
+                attachment.type = file.type;
+
+                // TODO: Figure out how best to check/limit file size pre-attachment...
+                if (file.size) { attachment.size = file.size; }
+                if (file.name) { attachment.download = file.name; }
+
+                var fileReader = new window.FileReader();
+                fileReader.readAsDataURL(file);
+                fileReader.onloadend = function (e) {
+                    if (e.target.result){
+                        attachment.href = e.target.result;
+                    } else {
+                        alert('ERROR: There was a problem reading the given file. Please try again.');
+                        return;
+                    }
+                }.bind(this);
+
+                fileArr.push(attachment);
+            }
+
+            // Concat with previous files
+            const allFiles = currFiles.concat(fileArr);
+
+            // Filter out duplicates (based on just filename for now; may need more criteria in future)
+            const dedupedFiles = _.uniq(allFiles, false, (file) => file.download);
+
+            this.setState({
+                files: dedupedFiles
+            });
+        }
+    }
+
+    handleRemoveFile(filename) {
+        const { multiselect } = this.props;
+
+        if (multiselect) {
+            const { files } = this.state;
+
+            // Filter to remove the clicked file by name (assuming no duplicate filenames)
+            const newFiles = files.filter((file) => {
+                if ((file.download === filename)) {
+                    return false;
+                }
+                return true;
+            });
+
+            this.setState({ files: newFiles });
+        } else {
+            this.handleClearAllFiles();
+        }
+    }
+
+    handleClearAllFiles() {
+        this.setState({ files: [] });
+    }
+
+    /**
+     * Constructs payload from props. If this is a payload for PATCH request with attachment, set attachmentPresent to true.
+     * @param {object} file                  A single file object, equivalent to something in this.state.files[i]
+     * @param {boolean} attachmentPresent    Is this a PATCH request/are you uploading a file? If so, set this to yes.
+     *
+     * Note: Started updating to use file information to auto-generate an alias for objects on 4DN & CGAP submission view;
+     * this isn't necessary in the current non-SV implementation, so has been left in a half-working state until that
+     * functionality is needed. In non-SV cases on CGAP pedigreeviz, this will skip those conditionals.
+     */
+    generatePayload(file, attachmentPresent) {
+        const { award, lab, institution, project } = this.props;
+
+        const aliasFilename = file.download.split(' ').join('');
+        let alias;
+
+        const payloadObj = {};
+
+        // If on 4DN, use lab and award data (institution/project should be null)
+        if (lab && award) {
```

Comment:
(For future) Great for now; we should remove this logic here and in SubmissionView (and have it be done for us in backend). Gotta wait for backend to do this first though I think / we could check in w. Kent & Will later.